pkg/wireguard/wireguard.go

```diff
@@ -265,3 +314,69 @@ func (dev *Device) Close() error {
 
 	return nil
 }
+
+func createWireguardDevice(cli *wgctrl.Client, listenPort int, privateKey *wgtypes.Key) error {
+	c, err := rtnetlink.Dial(nil)
+	if err != nil {
+		return fmt.Errorf("error connecting to netlink: %w", err)
+	}
+
+	defer c.Close() //nolint:errcheck
+
+	err = c.Link.New(&rtnetlink.LinkMessage{
+		Family: unix.AF_UNSPEC,
+		Type:   unix.ARPHRD_NONE,
+		Attributes: &rtnetlink.LinkAttributes{
+			Name: interfaceName,
+			Type: 65534,
+			Info: &rtnetlink.LinkInfo{
+				Kind: linkKindWireguard,
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("error creating wireguard device: %w", err)
+	}
+
+	port := listenPort
+	wgconfig := wgtypes.Config{
```

Comment:
don't we do same in `configurePrivateKey()` method? it should work same way for userspace and in-kernel implementations

Reply:
Yes we do, I missed it somehow. Cleaned it up.